### ISSUE TITLE
possible false positive E303

### ISSUE THREAD

**Issue by dimaqq** [PyCQA/pycodestyle#168]
Consider following valid code, at module top level:
```python
# snip
class Foo:
    pass


class Bar:
    pass
# snip
```

Consider making it conditional, at module top level:
```python
# snip
if something:
    class Foo:
        pass


    class Bar:
        pass
# snip
```

pep8 reports E303 too many blank lines (2) before Bar in the conditional case.
semantically it is a false positive.
syntactically it is an unclear case.

from PEP 8:
Separate top-level function and class definitions with two blank lines.
Method definitions inside a class are separated by a single blank line.
Extra blank lines blah blah blah...

this particular case doesn't fall in either clearly defined category.

**Comment by jayvdb** [comment]
I inherited a bunch of conditional classes, which was a OS-dependent implementation as suggested by garyd203 , and were basically this recipe:

http://stackoverflow.com/questions/878972/windows-cmd-encoding-change-causes-python-crash/3259271#3259271

The fix was not to worry about pep8's limitations, but completely overhaul it to move the classes to the top-level.
https://gerrit.wikimedia.org/r/#/c/239540/3/pywikibot/userinterfaces/win32_unicode.py,cm

**Comment by IanLee1521** [comment]
Closing as I feel that since it doesn't fall into either of the "two line separation" cases, this should be handled with a single line of indentation between the classes. If there is a strong opinion, I would recommend reaching out to the python-dev list to try to get the PEP-8 document updated to clarify this case.
